### PR TITLE
pd_client: unwrap or panic with detail log

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Sometimes PD client panic with an unreadable message, and it is not helpful for debugging.

### What is changed and how it works?

Add a detail log when unwrapping a value fails.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

- Improve PD client log.

### CODE REVIEW

File: components/pd_client/src/client.rs [hunk 1]
```diff
@@ -325,7 +325,9 @@ impl PdClient for RpcClient {
                 .rl()
                 .client_stub
                 .get_region_by_id_async_opt(&req, Self::call_option())
-                .unwrap();
+                .unwrap_or_else(|e| {
+                    panic!("fail to request PD {} err {:?}", "get_region_by_id", e)
```
Comment (BusyJay) [inline components/pd_client/src/client.rs:329]: Why not inline "get_region_by_id"?

**Comment by overvenus** [comment]
Some panic is not obvious to me, consider this example. So I make it panics with an explicit message.

<details><summary>Backtrace</summary>

```
2020/06/01 01:11:26.147 +08:00] [ERROR] [panic_hook.rs:104] ["thread 'pd-worker' panicked 'called `Option::unwrap()` on a `None` value' at \"libcore/option.rs:345\"\nstack backtrace:
   0:     0x7f9b92f879ec - backtrace::backtrace::libunwind::trace::h102c40391864e832
                        at /rust/registry/src/github.com-1ecc6299db9ec823/backtrace-0.2.3/src/backtrace/libunwind.rs:54
                         - backtrace::backtrace::trace::h74b1dbace51da4c0
           at /rust/registry/src/github.com-1ecc6299db9ec823/backtrace-0.2.3/src/backtrace/mod.rs:70
   1:     0x7f9b928cabc9 - tikv::util::panic_hook::set_exit_hook::{{closure}}::h2ac833b427aa1e9e
                        at /home/jenkins/.target/release/build/backtrace-4ca9820cec07bbc1/out/capture.rs:79
   2:     0x7f9b930193cf - std::panicking::rust_panic_with_hook::h60a37fb79e459606
                        at libstd/panicking.rs:479
   3:     0x7f9b9301922e - std::panicking::continue_panic_fmt::hb542a58a3d52cb5c
                        at libstd/panicking.rs:390
   4:     0x7f9b9301c655 - rust_begin_unwind
         at libstd/panicking.rs:325
   5:     0x7f9b9302b252 - core::panicking::panic_fmt::h68c841fe431ac371
                        at libcore/panicking.rs:77
   6:     0x7f9b9302b20a - core::panicking::panic::hb6cf555efce2ed02
                        at libcore/panicking.rs:52
   7:     0x7f9b928a817b - <futures::future::chain::Chain<A, B, C>>::poll::h4de11ae2aa7162f8
                        at /checkout/src/libcore/macros.rs:20
                         - <futures::future::and_then::AndThen<A, B, F> as futures::future::Future>::poll::h36134db381c82013
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/and_then.rs:32
   8:     0x7f9b928a0026 - <alloc::boxed::Box<F> as futures::future::Future>::poll::h7bfc0c074ed8c9b4
           at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/mod.rs:113
                         - <futures::future::chain::Chain<A,B, C>>::poll::h4ae23b94eb6bbf80
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/chain.rs:26
   9:     0x7f9b928a6836 - <futures::future::then::Then<A, B, F> as futures::future::Future>::poll::hc09088fcb0dc5036
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/then.rs:32
                         - <futures::future::chain::Chain<A, B, C>>::poll::h95175ee2c36fac43
               at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/chain.rs:32
                         - <futures::future::and_then::AndThen<A, B, F> as futures::future::Future>::poll::h4649bb497d00abe0
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/and_then.rs:32
  10:     0x7f9b928a54e4 - <alloc::boxed::Box<F> as futures::future::Future>::poll::h9e100e873bbd0fdf
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/mod.rs:113
                         - <futures::future::chain::Chain<A, B, C>>::poll::h3594fd36462c0c5e
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/chain.rs:32
                         - <futures::future::and_then::AndThen<A, B, F> as futures::future::Future>::poll::hd353c9f928d177f8
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/and_then.rs:32
                         - <futures::future::chain::Chain<A, B, C>>::poll::h6b14770e6076020b
        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/chain.rs:26
  11:     0x7f9b92a6a1fa - <futures::future::then::Then<A, B, F> as futures::future::Future>::poll::h3857d8eba890a1e5
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/then.rs:32
  12:     0x7f9b931b2ff7 - <futures::future::map_err::MapErr<A, F> as futures::future::Future>::poll::hb7b92267e3ed5452
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/mod.rs:113
  13:     0x7f9b92be1e04 - <alloc::boxed::Box<F> as futures::future::Future>::poll::h8ac3e330b11b5392
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/mod.rs:113
 - <futures::task_impl::Spawn<T>>::poll_future_notify::{{closure}}::he001a172e31779ff
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:289
                         - <futures::task_impl::Spawn<T>>::enter::{{closure}}::hdee3bc717457addc
    at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:363
                         - futures::task_impl::std::set::h51ec6300245f43ea
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/std/mod.rs:78
                         - <futures::task_impl::Spawn<T>>::enter::he4fa56ab2cef6851
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:363
                         - <futures::task_impl::Spawn<T>>::poll_future_notify::h41e87b363f98d453
                        at /rust/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:289
                         - <tokio::executor::current_thread::Entered<'a, P>>::tick::h01ae2ad5f14e7fc9
                        at /rust/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/executor/current_thread/scheduler.rs:352
  14:     0x7f9b925a22c1 - tokio_core::reactor::Core::run::hd2ab252e3641bc0b
                        at /rust/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/executor/current_thread/mod.rs:537
  15:     0x7f9b92617110 - tikv::util::worker::future::poll::h4b6a170ed18b436d
                        at /home/jenkins/workspace/release_tidb_2.1/tikv/src/util/worker/future.rs:120
                         - <tikv::util::worker::future::Worker<T>>::start::{{closure}}::hb16c554b0f35ddfb
                at /home/jenkins/workspace/release_tidb_2.1/tikv/src/util/worker/future.rs:151
                         - std::sys_common::backtrace::__rust_begin_short_backtrace::h53dba8ffed6773a9
                        at /checkout/src/libstd/sys_common/backtrace.rs:136
  16:     0x7f9b9307e5fc - std::thread::Builder::spawn::{{closure}}::hf471e57a6fdc9e12
                        at /checkout/src/libstd/thread/mod.rs:409
                         - <F as alloc::boxed::FnBox<A>>::call_box::h27eca30ecb4ad824
                        at /checkout/src/liballoc/boxed.rs:640
  17:     0x7f9b930072e3 - std::sys::unix::thread::Thread::new::thread_start::h96884314b12ba24c
                        at /checkout/src/liballoc/boxed.rs:650
  18:     0x7f9b91a2ae24 - start_thread
  19:     0x7f9b9154135c - clone
  20:                0x0 - <unknown>
```

</details>

File: components/pd_client/src/client.rs (hunk 1 above)
Comment (overvenus) [reply components/pd_client/src/client.rs:329]: I think they are the same.

Comment (BusyJay) [reply components/pd_client/src/client.rs:329]: I don't think they are the same. When searching for error, I normally paste the error message to quickly navigate to the location. Current format makes it impossible to search using "get_region_by_id err".

**Pull request merged**